.github/workflows/python-ci.yaml

```diff
@@ -0,0 +1,298 @@
+          nox-session: ${{ matrix.nox-session }}
+          git-username: ${{ secrets.GIT_USER }}
+          git-password: ${{ secrets.GIT_PASS }}
+
+  build:
+    name: Build distribution packages
+    needs: [test]
+    runs-on: ${{ matrix.platform.runner }}
+    strategy:
+      matrix:
+        platform:
+          - runner: ubuntu-latest
+            target: x86_64
+          - runner: ubuntu-latest
+            target: x86
+          - runner: ubuntu-latest
+            target: aarch64
+          - runner: ubuntu-latest
+            target: armv7
+          - runner: macos-14
+            target: x86_64
+          - runner: macos-14
+            target: aarch64
+          - runner: macos-15
+            target: x86_64
+          - runner: macos-15
+            target: aarch64
+        python:
+          - "3.11"
+          - "3.12"
+          - "3.13"
+    steps:
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
+      - name: Fetch sources
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON_VERSION }}
+          cache: 'pip'
+
+      - name: Install required Python packages
+        run: |
+          python -m pip install -U pip
+          python -m pip install -U build
+          pip freeze
+
+      - name: Build the source and binary distribution
+        run: python -m build
+
+      - name: Upload distribution files
+        uses: actions/upload-artifact@v4
+        with:
+          name: dist-packages-${{ matrix.platform.runner }}-${{ matrix.platform.target }}-${{ matrix.python }}
+          path: dist/
+          if-no-files-found: error
+
+  test-docs:
+    name: Test documentation website generation
+    if: github.event_name != 'push'
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup Git
+        uses: frequenz-floss/gh-action-setup-git@v0.x.x
+
+      - name: Fetch sources
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ env.DEFAULT_PYTHON_VERSION }}
+          cache: 'pip'
+
+      - name: Install build dependencies
+        run: |
+          python -m pip install -U pip
+          python -m pip install .[dev-mkdocs]
+          pip freeze
+
+      - name: Generate the documentation
+        env:
+          MIKE_VERSION: gh-${{ github.job }}
+        run: |
+          mike deploy $MIKE_VERSION
+          mike set-default $MIKE_VERSION
+
+      - name: Upload site
+        uses: actions/upload-artifact@v4
+        with:
+          name: docs-site
+          path: site/
+          if-no-files-found: error
+
+  publish-docs:
+    name: Publish documentation website to GitHub pages
+    if: startsWith(github.ref, 'refs/tags/v')
+    runs-on: ubuntu-20.04
+    needs: ["build", "test-docs"]
```

Comment:
This should not depend on `test-docs`, as it will not run now for pushes.

This is the only thing that really needs fixing, the other are optional.